config/v1/types_feature.go

```diff
@@ -28,6 +28,7 @@ type FeatureGate struct {
 	// spec holds user settable values for configuration
 	// +kubebuilder:validation:Required
 	// +required
+	// +kubebuilder:validation:XValidation:rule="has(oldSelf.featureSet) ? has(self.featureSet) : true",message=".spec.featureSet cannot be removed"
```

Comment:
if the new has the field, then we can count on the field validation to restrict the change.  If old has the field and new does not have the field, we fail generically rather than enumerate every case.